packages/core/admin/admin/src/content-manager/hooks/useDocument.ts

```diff
@@ -128,7 +132,16 @@ const useDocument: UseDocument = (args, opts) => {
         message: formatAPIError(error),
       });
     }
-  }, [toggleNotification, error, formatAPIError]);
+  }, [toggleNotification, error, formatAPIError, args.collectionType]);
+
+  React.useEffect(() => {
+    if (errorSchema) {
+      toggleNotification({
+        type: 'warning',
+        message: formatAPIError(errorSchema),
+      });
+    }
+  }, [toggleNotification, errorSchema, formatAPIError]);
 
```

Comment:
> why does one effect need formatAPIError as a dep and not the other?

They both have it :)

> why does one need args.collectionType

In single-types if we get a 404 it means we've not created the document for it yet. It's a bit of a weird API quirk we discussed this morning to be honest, because on the other side you can't "create" a single-type you have to "update" it 😞

Reply:
I guess I'm blind 🙃